unique_sdk/unique_sdk/_api_resource.py

```diff
@@ -194,5 +207,14 @@ async def _static_request_async(
 
         requestor = APIRequestor(user_id=user_id, company_id=company_id)
 
-        response = await requestor.request_async(method_, url_, params)
-        return convert_to_unique_object(response, user_id, company_id, params)
+        max_retries = 3
+        attempts = 0
+        while attempts < max_retries:
+            try:
+                response = await requestor.request_async(method_, url_, params)
+                return convert_to_unique_object(response, user_id, company_id, params)
+            except APIError as e:
```

Comment:
please also make sure that only a select few errors are leading to retrys
like the one I iinked

Reply:
https://unique-ag.slack.com/archives/C05C41NKFHA/p1727966417258559?thread_ts=1727943831.801709&cid=C05C41NKFHA

Reply:
Yes, I actually checked this, it specifically raised 
      "unique_sdk._error.APIError: There was a problem proxying the request"
 
in the log.

although our errors are kind of general exceptions and not catching any specific one.